script.js

```diff
@@ -1,49 +1,82 @@
-const db_local = 'stays.json'
-const db_url = 'https://raw.githubusercontent.com/pamdayne/wind-d-bnb/main/db/stays.json'
+const DB_URL = "//raw.githubusercontent.com/pamdayne/wind-d-bnb/main/db/stays.json";
 
 async function fetchLocations() {
-	try {
-		let resp = await fetch(db_url)
-		return await resp.json()
-	} catch (err) {
-		console.error(err)
-	}
+  return fetch(DB_URL)
+    .then(function (resp) { return resp.json() })
+    .then(function (json) {
+      return json;
+    }).catch(function (error) {
+      console.error(error);
+    })
 }
 
+// HOW CAN WE WRITE THIS BETTER? :D Anymore I can improve on?
```

Comment:
I can give a few pointers, but really depends on how far you want to refactor.

Reply:
1. If I were to re-factor the renderList method, I'll start by separating out each 'section' into a function. Having comments is a good code smell that you can use for this. I think you've done a good job of this for featImg. 

I think the code could look something like...

```
// generate sections
const featImg = getFeatureImage(...)
const heading = getHeading(...)

// assemble unit card
unitDiv = setContentElement('div', '', 'unit')
unitDiv.append(hostDiv, typeDiv, rateDiv)
```

This would make it a bit easier to read imo.

Reply:
2. As a next step, you could use a template generator like [handlebars](https://handlebarsjs.com/guide/). This would make it easier to read. For me it makes getting an idea of how the JS maps to HTML.

Reply:
3. You can consider a library, but I think it's out of scope for this task

Reply:
Oooh very nice suggetions Nipuna. 
You did 'demonstrated' a bit on how you would re-write this.

I will attempt this! 💪